include/aws/common/mutex.h

```diff
@@ -30,14 +30,15 @@ struct aws_mutex {
 #else
     pthread_mutex_t mutex_handle;
 #endif
+    bool initialized;
```

Comment:
Isn't mutex_handle != 0/NULL enough to flag init?

Reply:
Same, AWS_MUTEX_INIT zeroes out the handle on Windows.